data/examples/declaration/data/gadt-syntax-out.hs

```diff
@@ -1,3 +1,2 @@
 {-# LANUGAGE GADTSyntax #-}
-data Foo where
-  MKFoo :: a -> (a -> Bool) -> Foo
+data Foo where MKFoo :: a -> (a -> Bool) -> Foo
```

Comment:
I think we should keep the previous style because it's more readable (after all, that's the point of the tool!). Currently on `master` it works by re-formatting single version into the multi-line version without issues.

Reply:
In its current design, `ormolu` is not able to convert a single-line expression/statement to a multi-line one. The reason is that, since we format things differently on single line and multiple lines, if we re-format a single line-version to a multi-line one, there is a chance that we will format it differently on the next run, breaking the idempotency. As an example:


It works currently on `master` because we don't have a test where we have this problem in a nested scope. As an example, this wouldn't work on master (assuming we applied the fix for the closing bracket):

```
{-# LANGUAGE TemplateHaskell #-}

foo = [d|data Foo where MKFoo :: a -> (a -> Bool) -> Foo|]
```

Run 1:

```
{-# LANGUAGE TemplateHaskell #-}

foo = [d|data Foo where
    MKFoo :: a -> (a -> Bool) -> Foo
  |]
```

run 2:

```
{-# LANGUAGE TemplateHaskell #-}

foo = [d|
  data Foo where
    MKFoo :: a -> (a -> Bool) -> Foo
  |]
```

From ormolu's perspective, the `[d| |]` bracket in a single-line
layout on the "run 1", but on a multi-line layout on "run 2". 

There is another example of this issue in [this comment](https://github.com/tweag/ormolu/issues/258#issuecomment-519342454).

Reply:
This makes sense to me, thanks for the explanation.